integration/network/overlay/overlay_test.go

```diff
@@ -47,3 +54,57 @@ func TestEndpointWithCustomIfname(t *testing.T) {
 	assert.NilError(t, err)
 	assert.Assert(t, strings.Contains(out.Stdout, ": foobar@if"), "expected ': foobar@if' in 'ip link show':\n%s", out.Stdout)
 }
+
+// TestHostPortMappings checks that, when a Swarm task has ports mappings in
+// host mode, the ContainerList API endpoint reports the right number of ports.
+//
+// Regression test for
+func TestHostPortMappings(t *testing.T) {
+	skip.If(t, testEnv.IsRootless, "rootless mode doesn't support overlay networks")
+
+	ctx := setupTest(t)
+
+	d := daemon.New(t)
+	d.StartWithBusybox(ctx, t)
+	defer d.Stop(t)
+
+	d.SwarmInit(ctx, t, swarmtypes.InitRequest{AdvertiseAddr: "127.0.0.1:2377"})
+	defer d.SwarmLeave(ctx, t, true)
+
+	apiClient := d.NewClientT(t)
+
+	const netName = "testnet1"
+	network.CreateNoError(ctx, t, apiClient, netName,
+		network.WithDriver("overlay"),
+		network.WithAttachable())
+
+	svcID := swarm.CreateService(ctx, t, d,
+		swarm.ServiceWithNetwork(netName),
+		swarm.ServiceWithEndpoint(&swarmtypes.EndpointSpec{
+			Ports: []swarmtypes.PortConfig{
+				{Protocol: swarmtypes.PortConfigProtocolTCP, TargetPort: 80, PublishedPort: 80, PublishMode: swarmtypes.PortConfigPublishModeHost},
+			},
+		}))
+	defer apiClient.ServiceRemove(ctx, svcID)
+
+	poll.WaitOn(t, swarm.RunningTasksCount(ctx, apiClient, svcID, 1), swarm.ServicePoll)
+
+	ctrs, err := apiClient.ContainerList(ctx, containertypes.ListOptions{})
```

Comment:
Should we filter this list by state (to be "running")? Not sure if likely, but swarm could create a task, and on failure, create a new one (leaving the old tasks / containers in "exited" state).

I was initially thinking; we could use `swarm.GetRunningTasks()` for this, but I don't think that gives us information about the ports that are mapped, so would require listing the tasks, then getting the container-iD from them to inspect the container.

Reply:
> Should we filter this list by state (to be "running")?

`ListOptions` is empty, so only running containers should already be returned.